modules/compatibility/src/test/java/org/apache/ignite/compatibility/testframework/junits/CompatibilityTestIgniteNodeRunner.java

```diff
@@ -0,0 +1,172 @@
+import org.apache.ignite.Ignition;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.util.GridJavaProcess;
+import org.apache.ignite.internal.util.lang.GridAbsPredicate;
+import org.apache.ignite.internal.util.typedef.X;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.lang.IgniteInClosure;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.multijvm.IgniteNodeRunner;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Runs Ignite node.
+ */
+public class CompatibilityTestIgniteNodeRunner extends IgniteNodeRunner {
+    /** */
+    private static final String IGNITE_CLOSURE_FILE = System.getProperty("java.io.tmpdir") +
+        File.separator + "igniteClosure.tmp_";
+
+    /**
+     * Starts {@link Ignite} with test's default configuration.
+     *
+     * Command-line arguments specification:
+     * <pre>
+     * args[0] - required - path to closure for tuning IgniteConfiguration before node startup;
+     * args[1] - required - name of the starting node;
+     * args[2] - required - id of the starting node;
+     * args[3] - required - id of a node for synchronization of startup. Must be equals
+     * to arg[2] in case of starting the first node in the Ignite cluster;
+     * args[4] - optional - path to closure for actions after node startup.
+     * </pre>
+     *
+     * @param args Command-line arguments.
+     * @throws Exception In case of an error.
+     */
+    public static void main(String[] args) throws Exception {
+        X.println(GridJavaProcess.PID_MSG_PREFIX + U.jvmPid());
+
+        X.println("Starting Ignite Node... Args=" + Arrays.toString(args));
+
+        if (args.length < 3) {
+            throw new IllegalArgumentException("At least four arguments expected:" +
+                " [path/to/closure/file] [ignite-instance-name] [node-uuid] [sync-node-uuid] [optional/path/to/closure/file]");
+        }
+
+        IgniteConfiguration cfg = CompatibilityTestsFacade.getConfiguration();
+
+        IgniteInClosure<IgniteConfiguration> cfgClos = readClosureFromFileAndDelete(args[0]);
+
+        cfgClos.apply(cfg);
+
+        final UUID id = UUID.fromString(args[2]);
+        final UUID syncId = UUID.fromString(args[3]);
+
+        // Ignite instance name and id must be set according to arguments
+        // it's used for nodes managing: start, stop etc.
+        cfg.setIgniteInstanceName(args[1]);
+        cfg.setNodeId(id);
+
+        final Ignite ignite = Ignition.start(cfg);
+
+        // If 'id' equals 'syncId' then the starting node is the first node and
+        // there was no need to check the join to topology.
+        if (!id.equals(syncId)) {
+            GridTestUtils.waitForCondition(new GridAbsPredicate() {
+                @Override public boolean apply() {
+                    boolean found = ignite.cluster().node(syncId) != null;
```

Comment:
adding awaitPartitionMapExchange() lets you wait for node join cluster and that topology version is updated on all nodes.

Reply:
There is an another goal: synchronization of testing framework between two separate JVM.